src/spi/mod.rs

```diff
@@ -3,6 +3,15 @@
 pub mod blocking;
 pub mod nb;
 
+/// 8-bit SPI Word (default)
+pub type U8 = u8;
+/// 9-bit SPI Word
+pub struct U9;
+/// 16-bit SPI Word
+pub struct U16;
+/// 18-bit SPI Word
+pub struct U18;
```

Comment:
How are these zero-sized types supposed to be used?  I think you'd need something more like

```rust
pub struct U9(pub u16);
pub type U16 = u16;
pub struct U18(pub u32);
```

Reply:
You are right it was stupid for me. But you suggestion will not work also.
It should be sumething like:
```rust
pub trait SpiWord {
   type Word;
}
pub struct U9;
impl SpiWord for U9 {
    type Word = u16;
}
```

Reply:
I see, this solution also works.  However, now there isn't any implicit type annotation anymore so when calling, for example, `Transfer::transfer()` you'll have to be explicit about the datatype.  This means you cannot call it like

```rust
spi.transfer(&buf);
```

but you'll have to call it as

```rust
Transfer::<U8>::transfer(&mut spi, &buf);
```

My suggestion was to have newtype wrappers around the odd bitwidths and then store an array of those.  This would allow the compiler to infer the type automatically:

```rust
let buf: [U9; 16] = [U9(0), U9(23), ...];
spi.transfer(&buf);
```

Reply:
Wrapping each element of array is a pain